piper/buttondialog.py

```diff
@@ -96,13 +97,26 @@ def _init_ui(self, buttons):
         self.listbox.set_header_func(self._listbox_header_func)
 
         i = 0
-        for button in buttons:
-            key, name = self._get_button_name_and_description(button)
-            row = ButtonRow(name, _("Button mapping"), RatbagdButton.ACTION_TYPE_BUTTON, button.index + 1)
-            self.listbox.insert(row, i)
-            if self._action_type == RatbagdButton.ACTION_TYPE_BUTTON and button.index + 1 == self._button.mapping:
-                self.listbox.select_row(row)
+        # TODO: make this self._button.type == RatbagdButton.TYPE_LEFT or
+        # or self._button.type == RatbagdButton.TYPE_RIGHT once
+        # https://github.com/libratbag/libratbag/issues/233 is fixed.
+        if self._button.index == 0 or self._button.index == 1:
+            self.row_left_hand_mode._section = _("Button mapping")
+            self.listbox.insert(self.row_left_hand_mode, i)
+            if self._action_type == RatbagdButton.ACTION_TYPE_BUTTON and \
+                    (self._button.index + 2 == self._button.mapping or
+                        self._button.index == self._button.mapping):
```

Comment:
I wanted to add a comment here but decided against it. Wrong choice, hehe.

This `if` is there to highlight the current row if this option is currently set on the device. Since we're using the button's index as a replacement for its type (https://github.com/libratbag/libratbag/issues/233), we need to compare the index to the actual button mapping to see if it's a left/right mouse button assigned to left/right mouse click. Two scenarios:

1. Left mouse button (index 0) is mapped to right mouse button, where mappings are 1-indexed and thus right mouse click has value 2: `self._button.index + 2 == self._button.mapping`.
2. Right mouse button (index 1) is mapped to left mouse button, where mappings are again 1-indexed and thus left mouse click has value 1: `self._button.index == self._button.mapping`.

If this code sticks around with the new dialog I'll add a comment :)